File: packages/dprint-plugin-typescript/src/parsing/utils/isPrefixSemiColonInsertionChar.ts
let prefixSemiColonInsertionChars: Set<string> | undefined;

export function isPrefixSemiColonInsertionChar(char: string) {
    return getPrefixSemiColonInsertionChars().has(char);
}

function getPrefixSemiColonInsertionChars() {
    if (prefixSemiColonInsertionChars == null) {
        // from: https://standardjs.com/rules.html#semicolons
        prefixSemiColonInsertionChars = new Set(["[", "(", "`", "+", "*", "/", "-", ",", "."]);
    }
    return prefixSemiColonInsertionChars;
}

File: crates/dprint-plugin-typescript/src/utils/is_prefix_semi_colon_insertion_char.rs
pub fn is_prefix_semi_colon_insertion_char(value: char) -> bool {
    match value {
        // from: https://standardjs.com/rules.html#semicolons
        '[' | '(' | '`' | '+' | '*' | '/' | '-' | ',' | '.' => true,
        _ => false,
    }
}
